### PR TITLE
feat(base): migrate default preview to `@sanity/ui`

### PR DESCRIPTION
**Pull request by hermanwikner**
### Description

Migrate default preview to `@sanity/ui` and styled-components. The legacy `DefaultPreview` should not be in use anywhere in the app – except in the storybook files which is no longer maintained.

### What to review

Ensure that the component works as before. List panes, search results and reference options are some places where this component is used.

### Notes for release

Migrate `DefaultPreview` to `@sanity/ui` and styled-components

### CODE REVIEW

**Review by ricokahler — APPROVED** [review]
this looks good to me 😎. i just had one suggestion about `forwardRef`

I had some more questions in there and then I realized that the component was more or less a copy/paste from the original. is there a reason that one wasn't deleted yet?

File: packages/@sanity/base/src/components/previews/defaultPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          <Skeleton style={{width: 35, height: 35}} radius={2} marginRight={3} animated />
+          <Stack space={2} flex={1}>
+            <TextSkeleton style={{maxWidth: 320}} radius={1} animated />
+            <TextSkeleton style={{maxWidth: 200}} radius={1} size={1} animated />
+          </Stack>
+        </>
+      )}
+      {!isPlaceholder && (
+        <>
+          {media !== false && media !== undefined && (
```
[the remark this answers is not in the document]
Comment (ricokahler) [reply packages/@sanity/base/src/components/previews/defaultPreview.tsx:56]: This was old code. Resolving the convo…

File: packages/@sanity/desk-tool/src/panes/documentsListPane/DocumentsListPane.js [hunk 2]
```diff
@@ -192,14 +192,17 @@ export default class DocumentsListPane extends React.PureComponent {
...
-    />
+    <Box paddingX={2} paddingBottom={0}>
+      <PaneItem
+        id={getPublishedId(item._id)}
+        layout={this.state.layout || this.props.defaultLayout || 'default'}
+        value={item}
+        icon={this.props.displayOptions.showIcons === false ? false : undefined}
+        schemaType={schema.get(item._type)}
+        isSelected={this.itemIsSelected(item)}
+        isActive
```
Comment (ricokahler) [inline packages/@sanity/desk-tool/src/panes/documentsListPane/DocumentsListPane.js:203]: what's the difference between `isSelected` and `isActive`? in this context, why is `isActive` hard-coded true?

File: packages/@sanity/desk-tool/src/tool/DeskToolPanes.js [hunk 3]
```diff
@@ -249,6 +249,7 @@ export default class DeskToolPanes extends React.Component {
                     isCollapsed={isCollapsed}
                     isSelected={i === panes.length - 1}
                     isClosable={siblingIndex > 0}
+                    isActive={i === panes.length - 2}
```
Comment (ricokahler) [inline packages/@sanity/desk-tool/src/tool/DeskToolPanes.js:252]: what's the significance of second to last for `isActive`?

**Review by ricokahler — COMMENTED** [review]
Still looking great! I had some questions in there + one concern in there regarding whether or not `@sanity/base` should be a peerDep or not. Besides that, still looking good 😎

I'll defer my approval to @rexxars for some guidance on the peerDep thing. I'm not sure what the right answer is.

File: packages/@sanity/base/src/components/previews/defaultPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    object-fit: cover;
+  }
+
+  svg {
+    display: block;
+    font-size: calc(24 / 16 * 0.75em);
```
[the remark this answers is not in the document]
Comment (hermanwikner) [reply packages/@sanity/base/src/components/previews/defaultPreview.tsx:29]: Styling from previous component, these calculations is not related to sanity ui!

File: packages/@sanity/dashboard/package.json [hunk 5]
```diff
@@ -20,6 +20,7 @@
   ],
   "dependencies": {
     "@sanity/image-url": "^0.140.19",
+    "@sanity/base": "2.12.2",
```
[the remark this answers is not in the document]
Comment (vicbergquist) [reply packages/@sanity/dashboard/package.json:23]: hm yeah. Also I see that in multiple places it's both peer, dev and regular dependencies 🤔

File: packages/@sanity/desk-tool/src/panes/documentsListPane/DocumentsListPane.js (hunk 2 above)
Comment (vicbergquist) [reply packages/@sanity/desk-tool/src/panes/documentsListPane/DocumentsListPane.js:203]: Active pane items have different background and foreground 👍

File: packages/@sanity/dashboard/package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (rexxars) [reply packages/@sanity/dashboard/package.json:23]: Needs to be a peer + dev dependency

File: packages/@sanity/desk-tool/src/components/paneItem/PaneItem.js [hunk 6]
```diff
@@ -30,42 +65,46 @@ export default function PaneItem(props) {
...
-  return (
-    <PaneItemWrapper id={id} isSelected={isSelected} layout={layout} useGrid={useGrid}>
-      {preview}
-    </PaneItemWrapper>
+  const LinkComponent = useCallback(
+    (linkProps) => {
+      return <ChildLink {...linkProps} childId={id} />
+    },
+    [ChildLink, id]
   )
```
[the remark this answers is not in the document]
Comment (rexxars) [reply packages/@sanity/desk-tool/src/components/paneItem/PaneItem.js:87]: +1

**Review by rexxars — CHANGES_REQUESTED** [review]
Looks pretty good! Left a few comments inline.

From a brief comparison, it appears that the rendering of list panes are significantly heavier with the new UI components - I think we need to take a dive into the browser perf tools and see if we can figure out what's causing it. Could be something re-rendering too often, or it could be a side effect of more layers of UI being rendered somehow.

File: packages/@sanity/desk-tool/src/tool/DeskToolPanes.js (hunk 3 above)
Comment (mariuslundgard) [reply packages/@sanity/desk-tool/src/tool/DeskToolPanes.js:265]: The second-to-latest pane contains the currently active selection (the bright blue selection, as opposed to the inactive selection in the panes where index < panes.length - 2):
image

**Review by mariuslundgard — APPROVED** [review]
LGTM

File: packages/@sanity/base/src/__legacy/@sanity/components/previews/DefaultPreview.tsx [hunk 7]
```diff
@@ -39,6 +40,15 @@ const DEFAULT_MEDIA_DIMENSIONS: MediaDimensions = {
 
 class DefaultPreview extends React.PureComponent<DefaultPreviewProps> {
   // eslint-disable-next-line complexity
+
+  // eslint-disable-next-line class-methods-use-this
+  componentDidMount() {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'DefaultPreview from `part:@sanity/components/previews/default` is deprecated. Please import from `@sanity/base/components` instead'
```
[the remark this answers is not in the document]
Comment (hermanwikner) [reply packages/@sanity/base/src/__legacy/@sanity/components/previews/DefaultPreview.tsx]: Fixed, thanks! 👍

**Pull request merged**